seccon23-rev-perfect-blu/index.md

```diff
@@ -0,0 +1,403 @@
+---
+title: "seccon23 rev/Perfect Blu"
+publishDate: "18 Sep 2023"
+description: "Author: es3n1n"
+tags: ["rev", "seccon23"]
+---
+
+
+#### Description
+
+Perfect Blu (135 pt)
+
+No, I'm real!
+
+[perfect-blu.tar.gz](https://score.quals.seccon.jp/api/download?key=quals2023%2FPerfect_Blu%2Fperfect-blu.tar.gz) 367b6ed67dda0afbbc975ee70ee946b4c7bf9268
+
+
+#### Overview
+
+Once I opened the downloaded `.tar.gz` file I observed that there was a `.iso` file inside and once I mounted it I saw that it had a file structure that looked 
+like the fs structure of a DVD. So what I did is I dragged this `.iso` file to the VLC and was very surprised.
+
+![1](./img/1.png)
+
+As you probably already figured, this task is a DVD image that has an interactive menu where you need to enter the flag, press on the `CHECK` button and, depending on the flag it would, either show a `SUCCESS` or a `WRONG...` message.
+
+#### Analysing
+
+_I'm gonna be completely honest, I never had any experience with all this DVD stuff before, so I had to spend a couple of hours googling how to analyse this stuff first._
+
+There a multiple types of DVD menus that I'm aware of:
+- Java menus
+- IGS Menus
+
+What I did first is I checked for any .jar files or just something that could remind me of Java, but I didn't find anything and it meant that I was dealing with the IGS menu.
+
+The first step in IGS menu analysis is to download [BDEdit](https://bdedit.pel.hu/) and open up our mounted `BDMV\index.bdmv`
+
+![2](./img/2.png)
+
+Looks scary, I know. What I did is, I checked the clip info for the `und` streams. To do this I clicked on the `CLIPINFO` menu and found the `und` stream in the GUI that contained our buttons logic.
+
+![3](./img/3.png)
+
+Here, at the left top corner, there's a combo box with a clip selector, there are 96 clips and all of them have some buttons. 
+
+When I double-clicked at the `und` stream, the menu with buttons opened up and I saw a lot of buttons and the disasm of the code that they were doing.
+
+![4](./img/4.png)
+
+By messing around and guessing I found that the first default valid button(`1FDE` on the screenshot) contains some random stuff that I wasn't interested in, and I should check others instead.
+
+![13](./img/13.png)
+
+In this bytecode there's a `Call Object **` instruction with opcode `21820000` and only one operand, that just starts playing another menu that you provide as its first operand. Knowing this, I started analysing what the other buttons were doing.
+
+
+By observing all the other buttons I saw three types of buttons.
+
+---
+
+Type 1, what most buttons are doing.
```

Comment:
maybe would be better if you named them somewhat more meaningful? like "JumpTo48" instead of just "Type 1"

_what most buttons are doing._ - sounds meh, maybe remove `doing`? also what _most_?

also maybe this would better be a level3 header than a line separator (but this will require re-doing the structure of the document)

also i don't see much point in splitting the name and explanation with a picture, so maybe

```
**JumpTo48** - most of the buttons jump from the current menu (0) to menu 48

![6](./img/6.png)
```

also what is menu 48? you described menu 96 (the WRONG one), but not 48

same applies to the following "types" as well